stable/mongodb-replicaset/README.md

```diff
@@ -89,6 +92,72 @@ parameter `auth.enabled`. Once enabled, keyfile access control is set up and an
 is created. User credentials and keyfile may be specified directly. Alternatively, existing secrets may be provided. 
 The secret for the admin user must contain the keys `user` and `password`, that for the key file must contain `key.txt`.
 
+## TLS support
+
+To enable full TLS encryption setl `tls.enabled` to `true`. It is recommended to create your own CA by executing:
+
+```console
+$ openssl genrsa -out ca.key 2048
+$ openssl req -x509 -new -nodes -key ca.key -days 10000 -out ca.crt -subj "/CN=mydomain.com"
+```
+
+After that paste the base64 encoded (`cat ca.key | base64 -w0`) cert and key into the fields `tls.cacert` and
+`tls.cakey`. Adapt the configmap for the replicaset as following:
+
+```yml
+configmap:
+  storage:
+    dbPath: /data/db
+  net:
+    port: 27017
+    ssl:
+      mode: requireSSL
+      CAFile: /ca/tls.crt
+      clusterFile: /work-dir/mongo.pem
+      PEMKeyFile: /work-dir/mongo.pem
+  replication:
+    replSetName: rs0
+  security:
+    authorization: enabled
+    clusterAuthMode: x509
+    keyFile: /keydir/key.txt
+```
+
+To access the cluster you need one of the certificates generated during cluster setup in `/work-dir/mongo.pem` or you
```

Comment:
You reference `/work-dir/mongo.pem` here as if it were hard-coded, but it is configurable. Maybe it shouldn't even be configurable, since it is generated. But then, you always generated the certificates. I think it should be possible that the user provide their own certificates so they don't have to be generated.

Reply:
Hm, the user already can give a certificate (CA) within the config for which the cluster certs will be signed. Giving own certificates depends highly on the hostnames and the amount of replicas. I am not sure how to implement this, and I guess for now it is more convenient to let the init container create their own certificates and simply sign a client cert for external usage.

Reply:
Fair enough.